services/SchedulableTransactionBody.proto

```diff
@@ -0,0 +1,114 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+option java_multiple_files = true;
+
+import "SystemDelete.proto";
+import "SystemUndelete.proto";
+import "Freeze.proto";
+
+import "ContractCall.proto";
+import "ContractCreate.proto";
+import "ContractUpdate.proto";
+
+import "CryptoCreate.proto";
+import "CryptoDelete.proto";
+import "CryptoTransfer.proto";
+import "CryptoUpdate.proto";
+
+import "FileAppend.proto";
+import "FileCreate.proto";
+import "FileDelete.proto";
+import "FileUpdate.proto";
+
+import "ContractDelete.proto";
+
+import "ConsensusCreateTopic.proto";
+import "ConsensusUpdateTopic.proto";
+import "ConsensusDeleteTopic.proto";
+import "ConsensusSubmitMessage.proto";
+
+import "TokenCreate.proto";
+import "TokenFreezeAccount.proto";
+import "TokenUnfreezeAccount.proto";
+import "TokenGrantKyc.proto";
+import "TokenRevokeKyc.proto";
+import "TokenDelete.proto";
+import "TokenUpdate.proto";
+import "TokenMint.proto";
+import "TokenBurn.proto";
+import "TokenWipeAccount.proto";
+import "TokenAssociate.proto";
+import "TokenDissociate.proto";
+
+import "ScheduleDelete.proto";
+
+/* A schedulable transaction. Note that the global/dynamic system property 
+<tt>scheduling.whitelist</tt> controls which transaction types may be scheduled. 
+In Hedera Services 0.13.0, it will include only <tt>CryptoTransfer</tt> and
+<tt>ConsensusSubmitMessage</tt> functions. */
+message SchedulableTransactionBody {
+  uint64 transactionFee = 1; // The maximum transaction fee the client is willing to pay
+  string memo = 2; // A memo to include the execution record; the UTF-8 encoding may be up to 100 bytes and must not include the zero byte
+  oneof data {
```

Comment:
There is no inherit problem with this, but I wanted to mention that by naming fields inside of a `oneof` in camelCase instead of lower_snake_case it makes it harder to update these fields later since it breaks the generated Java protobuf code. Java generates a `DataCase` enum with values `CONTRACTCALL` for `contractCall` and `CONTRACT_CALL` for `contract_call`. This is why in https://github.com/hashgraph/hedera-protobufs/pull/13 I wasn't able to rename fields inside a `oneof`. If that PR is to ever be merged (which from my understand it is), then it'd easier to update if these fields were renamed to lower_snake_case now, rather than later. If this cannot be done, then that is fine, but just wanted to make sure you're aware of  this issue.

Reply:
Ah I didn't know this; renamed to lower_snake_case!